contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -698,14 +699,14 @@ SetBulkLoadRowData(TDSRequestBulkLoad request, StringInfo message)
 							{
 								rowData->isNull[i] = true;
 								i++;
-								token->isNull = false;
+								FreePlpToken(token);
```

Comment:
1. why do we need to call FreePlpToken when we know certainly that it is PLP_NULL? 
2. why is `token->isNull = false` not needed anymore?

Reply:
> 1. why do we need to call FreePlpToken when we know certainly that it is PLP_NULL?

`ParameterTokenData` struct itself needs to be freed in this case to not leak it. It is indeed not necessary to call `FreePlpToken` on it, `pfree(token)` would be enough. `FreePlpToken` is used here for consistency, it works correctly on empty token.

>     2. why is `token->isNull = false` not needed anymore?

AFAICS this flag (and the token struct itself) is not accessed after that point - token struct is [allocated again](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/4a1e451fcfd0387aa2479a1ee3605130c49e2c94/contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c#L692) before the next use. So we are freeing the token in this case instead of resetting `isNull` flag.

Reply:
Makes sense! Thanks for clarification